app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb

```diff
@@ -183,30 +182,37 @@ def create_managed_virtual_machine(vm_hash)
           vm.hardware_profile = ComputeModels::HardwareProfile.new.tap do |hw_profile|
             hw_profile.vm_size = vm_hash[:vm_size]
           end
-          vm.network_profile = define_network_profile(vm_hash[:network_interface_card_ids])
         end
 
-        response = sdk.create_or_update_vm(vm_hash[:resource_group], vm_hash[:name], vm_create_params)
+        vm_create_params
+      end
+
+      def create_managed_virtual_machine(vm_hash)
+        vm_params = initialize_vm(vm_hash)
+        vm_params.network_profile = define_network_profile(vm_hash[:network_interface_card_ids])
+
+        actual_vm = sdk.create_or_update_vm(vm_hash[:resource_group], vm_hash[:name], vm_params)
         logger.debug "Virtual Machine #{vm_hash[:name]} Created Successfully."
-        response
+        actual_vm
```

Comment:
Instead of lines([194..196](https://github.com/theforeman/foreman_azure_rm/pull/43/files#diff-01cbc39b299d0b7077bd73999f39a4caR194-R196)), simply return -

`sdk.create_or_update_vm(vm_hash[:resource_group], vm_hash[:name], vm_params)`

Reply:
done.